dspace-api/src/main/java/org/dspace/statistics/util/IPTable.java

```diff
@@ -45,155 +79,131 @@ public class IPTable {
 
-        String[] end;
+        String end;
 
         String[] range = ip.split("-");
 
-        if (range.length >= 2) {
+        if (range.length == 2) {
 
-            start = range[0].trim().split("/")[0].split("\\.");
-            end = range[1].trim().split("/")[0].split("\\.");
-
-            if (start.length != 4 || end.length != 4) {
-                throw new IPFormatException(ip + " - Ranges need to be full IPv4 Addresses");
-            }
+            start = range[0].trim();
+            end = range[1].trim();
 
-            if (!(start[0].equals(end[0]) && start[1].equals(end[1]) && start[2].equals(end[2]))) {
-                throw new IPFormatException(ip + " - Ranges can only be across the last subnet x.y.z.0 - x.y.z.254");
+            try {
+                long ipLo = ipToLong(InetAddress.getByName(start));
+                long ipHi = ipToLong(InetAddress.getByName(end));
+                ipRanges.add(new IPRange(ipLo, ipHi));
+                return;
+            } catch (UnknownHostException e) {
+                throw new IPFormatException(ip + " - Range format should be similar to 1.2.3.0-1.2.3.255");
             }
 
         } else {
-            //need to ignore CIDR notation for the moment.
-            //ip = ip.split("\\/")[0];
-
-            String[] subnets = ip.split("\\.");
-
-            if (subnets.length < 3) {
-                throw new IPFormatException(ip + " - require at least three subnet places (255.255.255.0");
-            }
-
-            start = subnets;
-            end = subnets;
-        }
-
-        if (start.length >= 3) {
-            Map<String, Map<String, Set<String>>> first = map.get(start[0]);
-
-            if (first == null) {
-                first = new HashMap<>();
-                map.put(start[0], first);
-            }
-
-            Map<String, Set<String>> second = first.get(start[1]);
-
-            if (second == null) {
-                second = new HashMap<>();
-                first.put(start[1], second);
-            }
-
-            Set<String> third = second.get(start[2]);
-
-            if (third == null) {
-                third = new HashSet<>();
-                second.put(start[2], third);
+            // Convert implicit ranges to netmask format
+            //  192       -> 192.0.0.0/8
+            //  192.168   -> 192.168.0.0/16
+            //  192.168.1 -> 192.168.1.0/24
+            int periods = StringUtils.countMatches(ip, '.');
+            if (periods < 3) {
+                ip = StringUtils.join(ip, StringUtils.repeat(".0", 4 - periods - 1), "/", (periods + 1) * 8);
             }
 
-            //now populate fourth place (* or value 0-254);
-
-            if (start.length == 3) {
-                third.add("*");
+            if (ip.contains("/")) {
+                String[] parts = ip.split("/");
+                try {
+                    byte[] octets = InetAddress.getByName(parts[0]).getAddress();
+                    long result = 0;
```

Comment:
long result = ipToLong(InetAddress.getByName(parts[0])); // instead of this duplication?

Reply:
@benbosman : waiting for you to see this comment before merging this PR.  I think @mwoodiupui is correct that you could remove code duplication here by replacing lines 115-120 in `IPTable` with simply:
```
long result = ipToLong(InetAddress.getByName(parts[0]));
```
It looks to be a nice simplification...so, I'd appreciate it if we could add this simplification before merging.  Thanks!